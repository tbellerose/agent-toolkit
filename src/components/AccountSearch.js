import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setToken, clearToken } from '../actions/auth';
import { startSetSites, clearSites } from '../actions/sites';

export class AccountSearch extends Component {
  state = {
    error: '',
    authToken: this.props.authToken
  };

  onAuthTokenChange = (e) => {
    let authToken = e.target.value;
    this.setState(() => ({
      authToken
    }));
  };

  onSearch = async () => {
    if (!this.state.authToken) {
      this.setState(() => ({
        error: 'Please provide an auth token'
      }));
    } else {
      await this.props.setToken(this.state.authToken);
      this.props.startSetSites();
    }
  };

  onClear = () => {
    this.setState(() => ({
      authToken: ''
    }));
    this.props.clearToken();
    this.props.clearSites();
  };

  render() {
    return (
      <div className="page-header">
        {!!this.state.error &&
          <p className="form__error">{this.state.error}</p>
        }
        <div className="page-header__content">
          <input
            className="text-input"
            type="text"
            value={this.state.authToken}
            onChange={this.onAuthTokenChange}
            placeholder="Auth Token"
            autoFocus
          />
          <button className="button" onClick={this.onSearch}>Search</button>
          <button className="button" onClick={this.onClear}>Clear</button>
        </div>
      </div>
    );
  };
};

const mapStateToProps = (state) => ({
  authToken: state.auth.token,
  sites: state.sites
});

const mapDispatchToProps = (dispatch) => ({
  setToken: (token) => dispatch(setToken(token)),
  clearToken: () => dispatch(clearToken()),
  startSetSites: () => dispatch(startSetSites()),
  clearSites: () => dispatch(clearSites())
});

export default connect(mapStateToProps, mapDispatchToProps)(AccountSearch);