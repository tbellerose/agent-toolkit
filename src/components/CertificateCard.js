import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import moment from 'moment';
import { getAPI } from '../utils/api';

export class CertificateCard extends Component {
  state = {
    domain: '',
    expireDate: null,
    status: '',
    error: '',
    ready: false
  };

  getCertificateInfo = async () => {
    const { authToken, site } = this.props;
    const response = await getAPI(`/sites/${site.id}/certificates`, authToken);
    const { domain, expireDate, status, error } = response[0];
    this.setState(() => ({
      domain,
      expireDate,
      status,
      error,
      ready: true
    }));
  };

  componentDidMount() {
    this.getCertificateInfo();
  }

  render() {
    const { domain, expireDate, status, error, ready } = this.state;
    return (
      <div className='card'>
        {error ? (
          <div className='card__content'>{error}</div>
        ) : (
          <div>
            {ready && (
              <div>
                <h3 className='card__title'>Certificate Info</h3>
                <div className='card__content'>
                  <p>Domain: {domain}</p>
                  <p>
                    Status:{' '}
                    {status === 'ISSUED' ? <span className='green'>{status}</span> : <span className='red'>{status}</span>}
                  </p>
                  <p>Exp. Date: {moment(expireDate).format('MM Do, YYYY')}</p>
                </div>
              </div>
            )}
          </div>
        )}
      </div>
    );
  }
}

CertificateCard.propTypes = {
  authToken: PropTypes.string
};

const mapStateToProps = state => ({
  authToken: state.auth.token
});

export default connect(mapStateToProps)(CertificateCard);
