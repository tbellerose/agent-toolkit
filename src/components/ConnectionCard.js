import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getAPI } from '../utils/api';

export class ConnectionCard extends Component {
  state = {
    hostname: '',
    port: undefined,
    status: '',
    error: '',
    ready: false
  };

  getConnectionInfo = async () => {
    const { authToken, site } = this.props;
    const { hostname, port, status, error } = await getAPI(
      `/support/sites/${site.id}/ssh`,
      authToken
    );
    this.setState(() => ({
      hostname,
      port,
      status,
      error,
      ready: true
    }));
  };

  componentDidMount() {
    this.getConnectionInfo();
  }

  render() {
    const { hostname, port, status, error, ready } = this.state;
    return (
      <div className="card">
        {!!error
          ? <div className="card__content">{error}</div>
          : (
            <div>
              {ready &&
                <div>
                  <h3 className="card__title">SSH/SFTP Info</h3>
                  <div className="card__content">
                    <p>Host: {hostname}</p>
                    <p>Port: {port}</p>
                    <p>Status: {this.state.status === 'active'
                      ? <span className="green">{status}</span>
                      : <span className="red">{status}</span>
                    }
                    </p>
                  </div>
                  <div className="card__action">
                    <button className="button">Create User</button>
                    <button className="button">Reset Permissions</button>
                  </div>
                </div>
              }
            </div>
          )
        }
      </div>
    );
  };
};

const mapStateToProps = (state) => ({
  authToken: state.auth.token
});

export default connect(mapStateToProps)(ConnectionCard);