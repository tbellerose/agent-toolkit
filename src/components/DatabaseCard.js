import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getAPI } from '../utils/api';
import { postAPI } from '../utils/api';
import MessageModal from './MessageModal';

export class DatabaseCard extends Component {
  state = {
    name: '',
    fqdn: '',
    port: undefined,
    adminUsername: '',
    adminPassword: '',
    displayMessageModal: false,
    messageModalText: '',
    error: '',
    ready: false
  };

  getDatabaseInfo = async () => {
    const siteId = this.props.site.id;
    const {
      name,
      fqdn,
      port,
      adminUsername,
      adminPassword,
      error
    } = await getAPI(`/sites/${siteId}/database`, this.props.authToken);

    this.setState(() => ({
      name,
      fqdn,
      port,
      adminUsername,
      adminPassword,
      error,
      ready: true
    }));
  };

  handleCloseMessageModal = () => {
    this.setState(() => ({ displayMessageModal: false }));
  };

  handleResetPassword = async () => {
    const siteId = this.props.site.id;
    const { error } = await postAPI(
      `/support/sites/${siteId}/database/resetPassword`,
      this.props.authToken
    );
    if (error) {
      this.setState(() => ({
        displayMessageModal: true,
        messageModalText: 'There was a problem resetting the password'
      }));
    } else {
      this.setState(() => ({
        displayMessageModal: true,
        messageModalText: 'Database password reset'
      }));
    }
  };

  componentDidMount() {
    this.getDatabaseInfo();
  };

  render() {
    const { name, fqdn, port, adminPassword, adminUsername, ready } = this.state;
    return (
      <div className="card">
        {!!this.state.error
          ? (
            <div className="card__content">
              <p>{this.state.error}</p>
            </div>
          ) : (
            <div>
              {ready &&
                <div>
                  <h3 className="card__title">Database Info</h3>
                  <div className="card__content">
                    <p>DB Name: {name}</p>
                    <p>FQDN: {fqdn}</p>
                    <p>Port: {port}</p>
                    <p>Username: {adminUsername}</p>
                    <p>Password: {adminPassword}</p>
                  </div>
                  <div className="card__action">
                    <button className="button" onClick={this.handleResetPassword}>Reset Password</button>
                    <button className="button">Kill Connections</button>
                  </div>
                  <MessageModal
                    modalIsOpen={this.state.displayMessageModal}
                    handleCloseModal={this.handleCloseMessageModal}
                    message={this.state.messageModalText}
                  />
                </div>
              }
            </div>
          )
        }
      </div>
    );
  };
};

const mapStateToProps = (state) => ({
  authToken: state.auth.token
});

export default connect(mapStateToProps)(DatabaseCard);