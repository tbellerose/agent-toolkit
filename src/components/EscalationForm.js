import React, { Component } from 'react';
import checkSites from '../utils/siteCheck';

export class EscalationForm extends Component {
  state = {
    siteId: this.props.site.id,
    primaryDomain: this.props.site.domains.primary.name,
    defaultDomain: this.props.site.domains.default.name,
    siteChecks: undefined,
    details: '',
    submitted: false,
    error: ''
  };

  handleSiteChecks = async () => {
    const { primaryDomain, defaultDomain } = this.state;
    let urls = [];
    if (primaryDomain === defaultDomain) {
      urls.push(primaryDomain);
    } else {
      urls.push(primaryDomain, defaultDomain)
    }
    const siteChecks = await checkSites(urls);
    this.setState(() => ({
      siteChecks
    }));
  };

  handleDetailsChange = (e) => {
    const details = e.target.value;
    this.setState(() => ({
      details
    }));
  };

  handleEscalate = async (e) => {
    e.preventDefault();
    this.handleSiteChecks();
    if (!this.state.details) {
      this.setState(() => ({
        error: 'Please provide request details'
      }));
    } else {
      this.setState(() => ({
        submitted: true,
        error: ''
      }));
    }
  };

  render() {
    const {
      siteId,
      primaryDomain,
      defaultDomain,
      details,
      siteChecks,
      submitted,
      error
    } = this.state;

    return (
      <div>
        {submitted
          ? (
            <div className="form__output">
              <p>#### MWP 2.0 Assistance Request ####</p>
              <p>Site ID: {siteId}</p>
              <p>Primary Domain: {primaryDomain}</p>
              <p>Default Domain: {defaultDomain}</p>
              <p>Details: {details}</p>
              <br />
              {siteChecks &&
                siteChecks.map((siteCheck, i) => (
                  <p key={i}>{siteCheck}</p>
                ))
              }
            </div>
          ) : (
            <form className="form">
              <p className="form__error--under">{error}</p>
              <div className="form__group">
                <label className="form__label">Site ID</label>
                <input
                  className="text-input"
                  type="text"
                  value={siteId}
                  readOnly
                />
              </div>
              <div className="form__group">
                <label className="form__label">Primary Domain</label>
                <input
                  className="text-input"
                  type="text"
                  value={primaryDomain}
                  readOnly
                />
              </div>
              <div className="form__group">
                <label className="form__label">Default Domain</label>
                <input
                  className="text-input"
                  type="text"
                  value={defaultDomain}
                  readOnly
                />
              </div>
              <div className="form__group">
                <label className="form__label">Details</label>
                <textarea
                  className="textarea"
                  value={details}
                  onChange={this.handleDetailsChange}
                  placeholder="Description of issue and troubleshooting steps"
                >
                </textarea>
              </div>
              <button className="button" onClick={this.handleEscalate}>Escalate</button>
            </form>
          )
        }
      </div>
    );
  };
};

export default EscalationForm;