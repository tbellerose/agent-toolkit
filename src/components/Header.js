import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { withAuth } from '@okta/okta-react';
import { clearUser, clearToken } from '../actions/auth';
import { clearSites } from '../actions/sites';

export class Header extends Component {
  state = { showMenu: false };

  toggleMenu = () => {
    this.setState((prevState) => ({
      showMenu: !prevState.showMenu
    }));
  };

  logout = async () => {
    this.props.clearToken();
    this.props.clearUser();
    this.props.clearSites();
    this.props.auth.logout('/');
  };

  render() {
    return (
      <header className='header'>
        <div className='header__content'>
          <Link className='header__title' to='/dashboard'>
            <h1>Agent Toolkit</h1>
          </Link>
          <div>
            <button className='button button--link' onClick={ this.toggleMenu }>
              {this.props.user && this.props.user.name}
              {this.state.showMenu
                ? <i className='arrow up'></i>
                : <i className='arrow down'></i>
              }
            </button>
            {this.state.showMenu &&
              <div className='dropdown-menu'>
                <button className='dropdown-menu__item' onClick={ this.logout }>Logout</button>
              </div>
            }
          </div>
        </div>
      </header>
    );
  }
}

Header.propTypes = {
  user: PropTypes.object,
  clearUser: PropTypes.func,
  clearToken: PropTypes.func,
  clearSites: PropTypes.func,
  auth: PropTypes.object
};

const mapStateToProps = (state) => ({
  user: state.auth.user
});

const mapDispatchToProps = (dispatch) => ({
  clearUser: () => dispatch(clearUser()),
  clearToken: () => dispatch(clearToken()),
  clearSites: () => dispatch(clearSites())
});

export default connect(mapStateToProps, mapDispatchToProps)(withAuth(Header));
