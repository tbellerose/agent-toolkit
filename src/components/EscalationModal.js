import React from 'react';
import Modal from 'react-modal';
import EscalationForm from './EscalationForm';

if (process.env.NODE_ENV !== 'test') {
  Modal.setAppElement('#app');
}

export const EscalationModal = (props) => (
  <Modal
    isOpen={props.modalIsOpen}
    contentlabel="Escalation Template"
    onRequestClose={props.handleCloseModal}
    closeTimeoutMS={200}
    className="modal"
  >
    <h2 className="modal__title">Escalation Template</h2>
    <EscalationForm site={props.site} />
  </Modal>
);

export default EscalationModal;
