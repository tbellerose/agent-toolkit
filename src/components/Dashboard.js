import React, { Component } from 'react';
import { withAuth } from '@okta/okta-react';
import Header from './Header';

export class Dashboard extends Component {
  logout = async () => {
    this.props.auth.logout('/');
  };

  render() {
    return (
      <div>
        <Header />
      </div>
    );
  };
};

export default withAuth(Dashboard);