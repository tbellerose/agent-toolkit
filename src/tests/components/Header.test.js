import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

let wrapper, auth, userinfo;

beforeEach(() => {
  auth = {
    logout: jest.fn(),
    isAuthenticated: jest.fn(),
  };
  userinfo = {
    name: 'Test User'
  }
  wrapper = shallow(<Header auth={auth} />);
  wrapper.setState(() => ({ authenticated: true, userinfo }))
});

test('should correctly render Header', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should update showMenu', () => {
  wrapper.find('button').at(0).simulate('click');
  expect(wrapper.state('showMenu')).toBe(true);
});

test('should call auth.logout', async () => {
  await wrapper.instance().logout();
  expect(auth.logout).toHaveBeenCalled();
});