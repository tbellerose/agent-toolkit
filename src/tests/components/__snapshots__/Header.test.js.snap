// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Header 1`] = `
<header
  className="header"
>
  <div
    className="header__content"
  >
    <Link
      className="header__title"
      replace={false}
      to="/dashboard"
    >
      <h1>
        Agent Toolkit
      </h1>
    </Link>
    <div>
      <button
        className="button button--link"
        onClick={[Function]}
      >
        Test User
        <i
          className="arrow down"
        />
      </button>
    </div>
  </div>
</header>
`;
